attrs.go

```diff
@@ -86,5 +100,27 @@ func fileStatFromInfo(fi os.FileInfo) (uint32, *FileStat) {
 	// os specific file stat decoding
 	fileStatFromInfoOs(fi, &flags, fileStat)
 
+	// The call above will include the sshFileXferAttrUIDGID in case
+	// the os.FileInfo can be casted to *syscall.Stat_t on unix.
+	// If implementations of RequestServers Handlers want to implement
+	// setting uids and gids while running on non-unix systems or
+	// without using *syscall.Stat_t underneath on unix, we offer
+	// an alternative way to support this through the FileInfoUidGid
+	// interface.
```

Comment:
So, I think my only remaining criticism, as small as it is, is that I was less worried about us internally tracking the proper usage of these extension `FileInfo` interfaces. That’s no problem, we maintain the code, and can “just get” how this works.

My concern was for importing code, and users of our package to know how and when the FileInfo extension interfaces should be used and what they will do. This means the documentation probably has to go into that for [`ListerAt`](https://pkg.go.dev/github.com/pkg/sftp#ListerAt) or [`FileLister`](https://pkg.go.dev/github.com/pkg/sftp#FileLister) right?

Reply:
Right. I'll try to move the documentation of this.